seatunnel-connectors-v2/connector-doris/src/main/java/org/apache/seatunnel/connectors/doris/datatype/AbstractDorisTypeConverter.java

```diff
@@ -0,0 +1,423 @@
+                    builder.dataType(BasicType.BOOLEAN_TYPE);
+                } else {
+                    builder.dataType(BasicType.BYTE_TYPE);
+                }
+                break;
+            case DORIS_SMALLINT:
+                builder.dataType(BasicType.SHORT_TYPE);
+                break;
+            case DORIS_INT:
+                builder.dataType(BasicType.INT_TYPE);
+                break;
+            case DORIS_BIGINT:
+                builder.dataType(BasicType.LONG_TYPE);
+                break;
+            case DORIS_FLOAT:
+                builder.dataType(BasicType.FLOAT_TYPE);
+                break;
+            case DORIS_DOUBLE:
+                builder.dataType(BasicType.DOUBLE_TYPE);
+                break;
+            case DORIS_CHAR:
+            case DORIS_VARCHAR:
+                if (typeDefine.getLength() != null && typeDefine.getLength() > 0) {
+                    builder.columnLength(typeDefine.getLength());
+                }
+                builder.dataType(BasicType.STRING_TYPE);
+                break;
+            case DORIS_LARGEINT:
+                builder.dataType(BasicType.STRING_TYPE);
+                builder.columnLength(MAX_DORIS_LARGEINT_TO_VARCHAR_LENGTH);
+                break;
+            case DORIS_STRING:
+            case DORIS_JSON:
+                builder.dataType(BasicType.STRING_TYPE);
+                builder.columnLength(MAX_STRING_LENGTH);
+                break;
+            default:
+                throw CommonError.convertToSeaTunnelTypeError(
+                        DorisConfig.IDENTIFIER, dorisColumnType, typeDefine.getName());
+        }
+    }
+
+    protected void sampleReconvertString(
+            Column column, BasicTypeDefine.BasicTypeDefineBuilder builder) {
+        // source is doris too.
+        if (column.getSourceType() != null
+                && column.getSourceType().equalsIgnoreCase(DORIS_LARGEINT)) {
+            builder.columnType(DORIS_LARGEINT);
+            builder.dataType(DORIS_LARGEINT);
+            return;
+        }
+
+        if (column.getColumnLength() == null || column.getColumnLength() <= 0) {
+            builder.columnType(DORIS_STRING);
+            builder.dataType(DORIS_STRING);
+            return;
+        }
+
+        if (column.getColumnLength() < POWER_2_8) {
+            if (column.getSourceType() != null
+                    && column.getSourceType().toUpperCase(Locale.ROOT).startsWith(DORIS_VARCHAR)) {
+                builder.columnType(
+                        String.format("%s(%s)", DORIS_VARCHAR, column.getColumnLength()));
+                builder.dataType(DORIS_VARCHAR);
+            } else {
+                builder.columnType(String.format("%s(%s)", DORIS_CHAR, column.getColumnLength()));
+                builder.dataType(DORIS_CHAR);
+            }
+            return;
+        }
+
+        if (column.getColumnLength() < POWER_2_16) {
```

Comment:
`POWER_2_16` not equals `65533`.
```suggestion
        if (column.getColumnLength() < 65533) {
```

Reply:
https://doris.apache.org/docs/sql-manual/sql-reference/Data-Types/VARCHAR/